build-service.py

```diff
@@ -88,73 +101,77 @@ def load_config_file(filename):
-        if isinstance(service_conf, str):  # if string, its a filepath to load from
-            service_conf = load_config_file(f"{repo_dir}/{service_conf}")
-
-        # pull image before building to make sure its supported
-        with open(f"{repo_dir}/Dockerfile", "r") as df:
-            fromregex = re.compile(r"^FROM (.+:?.+)", re.IGNORECASE | re.MULTILINE)
-            fromname = fromregex.search(df.read()).groups()[0]
-
-        logging.info(f"pulling base image {fromname} to check architecture")
-
-        # plat.machine() arch string doesn't match docker's arch string
-        platform_mappings = {
-            "python": "docker",
-            "x86": "386",
-            "x86_64": "amd64",
-            "armv7l": "arm",
-            "aarch64": "arm64",
-        }
-        platform_str = (
-            f"{platform.system()}/{platform_mappings[platform.machine()]}".lower()
-        )
+            raise BadGitBranch(branch)
+
+
+def build_repo(repo_dir, branch, deploy_conf, service_conf):
+    # do work in temp dir
+    logging.debug(
+        f"building repo '{repo_dir}@{branch}' with deploy conf '{deploy_conf}' and service conf '{service_conf}'"
+    )
+    # pull image before building to make sure its supported
+    with open(f"{repo_dir}/Dockerfile", "r") as df:
+        fromregex = re.compile(r"^FROM (.+:?.+)", re.IGNORECASE | re.MULTILINE)
+        fromname = fromregex.search(df.read()).groups()[0]
+
+    logging.info(f"pulling base image {fromname} to check architecture")
+
+    # plat.machine() arch string doesn't match docker's arch string
+    platform_mappings = {
+        "python": "docker",
+        "x86": "386",
+        "x86_64": "amd64",
+        "armv7l": "arm",
+        "aarch64": "arm64",
+    }
+    platform_str = (
+        f"{platform.system()}/{platform_mappings[platform.machine()]}".lower()
+    )
 
-        fromdata = dclient.images.get_registry_data(fromname)
-        platforms = [
-            f"{p['os']}/{p['architecture']}" for p in fromdata.attrs["Platforms"]
-        ]
-        if platform_str not in platforms:
-            raise ArchNotSupported(fromdata.image_name, platform_str, platforms)
+    fromdata = dclient.images.get_registry_data(fromname)
+    platforms = [f"{p['os']}/{p['architecture']}" for p in fromdata.attrs["Platforms"]]
+    if platform_str not in platforms:
+        raise ArchNotSupported(fromdata.image_name, platform_str, platforms)
 
-        logging.info(f"base image {fromname} good, continuing build")
+    logging.info(f"base image {fromname} good, continuing build")
 
-        image_name = deploy_conf["spec"]["template"]["spec"]["containers"][0]["image"]
-        dclient.images.build(path=repo_dir, tag=image_name)
+    image_name = deploy_conf["spec"]["template"]["spec"]["containers"][0]["image"]
+    dclient.images.build(path=repo_dir, tag=image_name)
+    image = deploy_conf["spec"]["template"]["spec"]["containers"][0]["image"]
+    dclient.images.build(path=repo_dir, tag=image)
```

Comment:
This got duplicated:
```suggestion
    image_name = deploy_conf["spec"]["template"]["spec"]["containers"][0]["image"]
    dclient.images.build(path=repo_dir, tag=image_name)
```